src/UserForm.test.js

```diff
@@ -0,0 +1,224 @@
+          {children}
+        </select>
+      </div>
+    );
+  },
+}));
+
+jest.mock('react-final-form', () => ({
+  Field: ({ name, label, component, onChange, required, validate, children }) => {
+    const Component = component;
+    return (
+      <div>
+        <label htmlFor={name}>{label}</label>
+        <Component
+          id={name}
+          name={name}
+          onChange={onChange}
+          required={required}
+          validate={validate}
+        >
+          {children}
+        </Component>
+      </div>
+    );
+  },
+}));
+
+jest.mock('@folio/stripes/smart-components', () => ({
+  ProxyManager: () => (
+    <div data-testid="proxy-manager" />
+  ),
+}));
+
+jest.mock('./utils', () => ({
+  ...jest.requireActual('./utils'),
+  getFullName: jest.fn(() => ({ user: '12', proxy:'11' })),
+  userHighlightBox: jest.fn(() => 'userHighlightBox'),
+}));
+
+jest.mock('./constants', () => ({
+  requestStatuses: jest.fn(() => ({ AWAITING_DELIVERY: 'AWAITING_DELIVERY', AWAITING_PICKUP:'AWAITING_PICKUP' })),
+}));
+
+jest.mock('react-intl', () => {
+  return {
+    ...jest.requireActual('react-intl'),
+    FormattedMessage: jest.fn(({ id, children }) => {
+      if (children) {
+        return children([id]);
+      }
+      return id;
+    }),
+  };
+});
+
+const user = {
+  id: '1234',
+  firstName: 'test first name',
+  lastName: 'test last name',
+  barcode: '123456',
+};
+const deliveryLocations = [
+  { value: 'location1', label: 'Location 1' },
+  { value: 'location2', label: 'Location 2' },
+];
+const fulfilmentTypeOptions = [
+  { labelTranslationPath: 'ui-requests.fulfilmentTypes.delivery', value: 'delivery' },
+  { labelTranslationPath: 'ui-requests.fulfilmentTypes.pickup', value: 'pickup' },
+];
+const servicePoints = [
+  { id: '1', name: 'Service Point 1' },
+  { id: '2', name: 'Service Point 2' },
+];
+const onChangeAddress = jest.fn();
+const onChangeFulfilment = jest.fn();
+const onCloseProxy = jest.fn();
+const onSelectProxy = jest.fn();
+
+describe('UserForm', () => {
+  it('renders delivery address select when deliverySelected is true', () => {
+    const defaultProps = {
+      deliveryLocations,
+      deliverySelected: true,
+      fulfilmentTypeOptions,
+      servicePoints,
+      user,
+      onChangeAddress,
+      onChangeFulfilment,
+      onCloseProxy,
+      onSelectProxy,
+      stripes: { connect: jest.fn((component) => component) },
+      patronGroup: 'Staff',
+      proxy: { id: '123', firstName: 'proxyFirstName', lastName: 'proxyLastName', barcode: '123456789' },
+      optionLabel: 'testoptionLabel',
+      fulfilmentPreference: 'fulfilmentPreference',
+      deliveryAddress: 'Location 1'
+    };
+    render(<UserForm {...defaultProps} />);
+    userEvent.selectOptions(
+      screen.getByRole('combobox', { name: 'ui-requests.deliveryAddress' }),
+      screen.getByRole('option', { name: 'Location 2' }),
+    );
+    expect(screen.getByRole('option', { name: 'Location 2' }).selected).toBe(true);
```

Comment:
Could you use data from constants created before instead of `{ name: 'Location 2' }`?
And could you go through all tests and use constants where it is possible?